src/storm/generator/JaniNextStateGenerator.cpp

```diff
@@ -267,61 +267,57 @@ namespace storm {
                 
                 STORM_LOG_DEBUG("Enumerated " << initialStateIndices.size() << " initial states using SMT solving.");
             } else {
-                CompressedState initialState(this->variableInformation.getTotalBitOffset(true));
                 
-                std::vector<int_fast64_t> currentIntegerValues;
-                currentIntegerValues.reserve(this->variableInformation.integerVariables.size());
-                for (auto const& variable : this->variableInformation.integerVariables) {
-                    STORM_LOG_THROW(variable.lowerBound <= variable.upperBound, storm::exceptions::InvalidArgumentException, "Expecting variable with non-empty set of possible values.");
-                    currentIntegerValues.emplace_back(0);
-                    initialState.setFromInt(variable.bitOffset, variable.bitWidth, 0);
+                // Create vectors holding all possible values
+                std::vector<std::vector<uint64_t>> allValues;
+                for (auto const& aRef : this->parallelAutomata) {
+                    auto const& aInitLocs = aRef.get().getInitialLocationIndices();
+                    allValues.template emplace_back(aInitLocs.begin(), aInitLocs.end());
+                }
+                uint64_t locEndIndex = allValues.size();
+                for (auto const& intVar : this->variableInformation.integerVariables) {
+                    STORM_LOG_ASSERT(intVar.lowerBound <= intVar.upperBound, "Expecting variable with non-empty set of possible values.");
+                    // The value of integer variables is shifted so that 0 is always the smallest possible value
+                    allValues.push_back(storm::utility::vector::buildVectorForRange<uint64_t>(static_cast<uint64_t>(0), intVar.upperBound + 1 - intVar.lowerBound));
```

Comment:
Could it make sense to only store the bounds and generate the range vector when it is needed? Storing the vectors directly may lead to large memory requirements.

Reply:
Yes, this would make sense. However, for locations, we do not have a consecutive list of ints and `storm::utility::combinatorics::forEach` requires that all iterators have the same type. 
There is probably some way around this (maybe using [boost range](https://www.boost.org/doc/libs/1_76_0/libs/range/doc/html/index.html) ) but to me, this seems to be not worth the effort. In particular, note that we'll have at least `allValues[0].size() * ... * allValues[n-1].size()` many states anyway which will most likely be the bottleneck.

Reply:
It might still cut the peak memory requirement by almost half as the vectors are kept in memory. Anyways, I understand that in practice this is unlikely to be a problem and probably not worth the effort.